pmd-core/src/main/java/net/sourceforge/pmd/cli/internal/commands/PMDCommand.java

```diff
@@ -0,0 +1,484 @@
+    public void setFormat(final String format) {
+        this.format = format;
+    }
+
+    @Option(names = { "--debug", "--verbose", "-D", "-V" }, description = "Debug mode.")
+    public void setDebug(final boolean debug) {
+        this.debug = debug;
+    }
+
+    @Option(names = { "--encoding", "-e" },
+            description = "Specifies the character set encoding of the source code files PMD is reading (i.e., UTF-8).",
+            defaultValue = "UTF-8")
+    public void setEncoding(final String encoding) {
+        this.encoding = encoding;
+    }
+
+    @Option(names = { "--benchmark", "-b" },
+            description = "Benchmark mode - output a benchmark report upon completion; default to System.err.")
+    public void setBenchmark(final boolean benchmark) {
+        this.benchmark = benchmark;
+    }
+
+    @Option(names = { "--stress", "-S" }, description = "Performs a stress test.")
+    public void setStress(final boolean stress) {
+        this.stress = stress;
+    }
+
+    @Option(names = { "--short-names" }, description = "Prints shortened filenames in the report.")
+    public void setShortnames(final boolean shortnames) {
+        this.shortnames = shortnames;
+    }
+
+    @Option(names = { "--show-suppressed" }, description = "Report should show suppressed rule violations.")
+    public void setShowSuppressed(final boolean showSuppressed) {
+        this.showSuppressed = showSuppressed;
+    }
+
+    @Option(names = { "--suppress-marker" },
+            description = "Specifies the string that marks a line which PMD should ignore.",
+            defaultValue = "NOPMD")
+    public void setSuppressMarker(final String suppressMarker) {
+        this.suppressMarker = suppressMarker;
+    }
+    
+    // TODO : "-min" is not a single letter option, but was never deprecated in PMD
+    // 6.x
+    @Option(names = { "--minimum-priority", "-min" },
+            description = "Rule priority threshold; rules with lower priority than configured here won't be used.%n"
+                    + "Valid values (case insensitive): ${COMPLETION-CANDIDATES}",
+            defaultValue = "Low")
+    public void setMinimumPriority(final RulePriority priority) {
+        this.minimumPriority = priority;
+    }
+
+    @Option(names = { "--property", "-P" }, description = "Key-value pair defining a property for the report format.")
+    public void setProperties(final Properties properties) {
+        this.properties = properties;
+    }
+
+    @Option(names = { "--report-file", "-r" },
+            description = "Path to a file to which report output is written. "
+                + "The file is created if it does not exist. "
+                + "If this option is not specified, the report is rendered to standard output.")
+    public void setReportFile(final Path reportFile) {
+        this.reportFile = reportFile;
+    }
+
+    // TODO : -version was never deprecated, should be replaced with --version… but
```

Comment:
Just a comment: https://github.com/pmd/pmd/issues/3426

Both `-language` and `-version` are meant to disappear

Reply:
Thanks for pointing to that issue, I was not tracking it. I'll implement that `--use-version` flag, it would be easier for autocompletion. Maybe I'll have to do a 1-arity (ie: `java-16` vs `java 16`), but let me experiment with that… I'm unsure how completion candidates work for arity > 1.

We still need to do the deprecations on PMD 6, this was a reminder to myself. My local branch had already removed these options